src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -42,6 +43,44 @@ public class InstanceManager : IInstanceManager
             _optionsFactory = optionsFactory ?? throw new ArgumentNullException(nameof(optionsFactory));
         }
 
+        public async Task<string> SpecializeMSISidecar(HostAssignmentContext context)
+        {
+            string msiEndpoint;
+            var msiEnabled = context.TryGetMsiEndpoint(out msiEndpoint);
+
+            _logger.LogInformation($"MSI enabled status: {msiEnabled}");
+
+            if (msiEnabled)
+            {
+                using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationMSIInit))
+                {
+                    var uri = new Uri(msiEndpoint);
+                    var address = $"http://{uri.Host}:{uri.Port}{ScriptConstants.LinuxMSISpecializationStem}";
```

Comment:
should this be https?

Reply:
this is the communication within the pod. only http seems to work. could be because of the way token service is configured or due to the pod config.

Reply:
Would be good to get to the bottom of that and determine whether we can use https